library/cohesity_clone_vm.py

```diff
@@ -0,0 +1,392 @@
+        clone_request = CloneTaskRequest()
+        clone_request.name = module.params.get('name')
+        clone_request.target_view_name = module.params.get('view_name')
+        clone_request.mtype = 'kCloneVMs'
+        clone_request.new_parent_id = protection_job_details.parent_source_id
+        clone_request.vmware_parameters = VmwareCloneParameters()
+        if module.params.get('suffix'):
+            clone_request.vmware_parameters.suffix = module.params.get('suffix')
+        if module.params.get('prefix'):
+            clone_request.vmware_parameters.prefix = module.params.get('prefix')
+        clone_request.vmware_parameters.powered_on = module.params.get(
+            'power_on')
+        clone_request.vmware_parameters.resource_pool_id = \
+            get_resource_pool_id(cohesity_client, module, resource_pool, protection_job_details.parent_source_id)
+        clone_request.objects = objects
+        clone_details = cohesity_client.restore_tasks.create_clone_task(
+            body=clone_request)
+        if not clone_details:
+            raise__cohesity_exception__handler("Failed to clone VMs", module)
+        status_message = wait(cohesity_client, module)
+        result = dict(
+            changed=True,
+            msg=status_message,
+            id=clone_details.id,
+            task_name=module.params.get('name')
+        )
+        module.exit_json(**result)
+    except APIException as ex:
+        raise__cohesity_exception__handler(
+            str(json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def destroy_clone(cohesity_client, module, clone_id):
+    '''
+    function to tear down clone
+    :param cohesity_client: cohesity client
+    :param module: object that holds parameters passed to the module
+    :param clone_id: clone task id
+    :return:
+    '''
+    try:
+        cohesity_client.restore_tasks.delete_public_destroy_clone_task(
+            id=clone_id)
+    except APIException as ex:
+        if "destroyed" in json.loads(ex.context.response.raw_body)['message']:
+            status = dict(
+                changed=False,
+                msg="Cohesity clone task is already destroyed",
+                task_name=module.params.get('name')
+            )
+            module.exit_json(**status)
+        else:
+            raise__cohesity_exception__handler(str(json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def get_cohesity_client(module):
+    '''
+    function to get cohesity cohesity client
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    cluster_vip = module.params.get('cluster')
+    username = module.params.get('username')
+    password = module.params.get('password')
+    domain = 'LOCAL'
+    if "/" in username:
```

Comment:
Is this consistent with other modules / too late to change?

Reply:
It is consistent with other modules, went by the initial design. May break backward compatibility if I change it now